### PR TITLE
ci: run sanitycheck with ninja

### PR DESCRIPTION
**Pull request by nashif**
This should speed things up a little bit.

We remove -R here due to some other breakage and use SANITYCHECK_OPTIONS again.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: .shippable.yml [hunk 1]
```diff
@@ -56,15 +56,15 @@ build:
             ./scripts/ci/get_modified_boards.py --commits origin/${PULL_REQUEST_BASE_BRANCH}..HEAD > modified_boards.args;
 
             if [ -s modified_boards.args ]; then
-              ./scripts/sanitycheck +modified_boards.args --save-tests test_file.txt;
+              ./scripts/sanitycheck ${SANITYCHECK_OPTIONS} +modified_boards.args --save-tests test_file.txt;
```
Comment (carlescufi) [inline .shippable.yml]: I assume sanitycheck already has support for Ninja with -N ?

File: .shippable.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ compiler: gcc
 env:
     global:
         - SDK=0.9.2
-        - SANITYCHECK_OPTIONS=" --inline-logs -R"
+        - SANITYCHECK_OPTIONS=" --inline-logs -N"
```
Comment (carlescufi) [inline .shippable.yml]: Why remove -R? Should this be part of a separate commit to disable asserts?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: .shippable.yml (hunk 1 above)
Comment (nashif) [reply .shippable.yml]: yes

File: .shippable.yml (hunk 2 above)
Comment (nashif) [reply .shippable.yml]: -R is broken right now, it was disable by mistake, see hunk below adding the options again, rather than hacking -N in, I am removing -R for now until the related issues are fixed and issue is open about this.

**Review by carlescufi — APPROVED** [review]
